### PR TITLE
refactor: clean up clip image encoder

### PR DESCRIPTION
**Pull request by bwanglzu**
1. polish readme
2. add more edge cases in tests.

### CODE REVIEW

File: jinahub/encoders/image/CLIPImageEncoder/README.md [hunk 1]
```diff
@@ -1,54 +1,68 @@
...
-can also choose to pass in already pre-processed images (see the class documentation).
+This encoder is meant to be used in conjunction with the [CLIPTextEncoder](https://github.com/jina-ai/executors/tree/main/jinahub/encoders/text/CLIPTextEncoder),
+as it can embedd text and images to the same latent space.
 
-The **CLIP** model was originally proposed in [Learning Transferable Visual Models From Natural Language Supervision](https://arxiv.org/abs/2103.00020), and is trained to embed images and text to the same latent
-space. The corresponding text encoder is **[CLIPTextEncoder](https://hub.jina.ai/executor/livtkbkg)**,
-using both encoders together works well in multi-modal or cross-modal search applications.
+The following arguments can be passed on initialization:
+
+- `pretrained_model_name_or_path`: name or path of the pre-trained CLIP model.
```
Comment (cristianmtr) [inline jinahub/encoders/image/CLIPImageEncoder/README.md:10]: We don't want the arguments in the README, right? They are parsed by Hubble and displayed in the hub.jina.ai interface

File: jinahub/encoders/image/CLIPImageEncoder/README.md [hunk 2]
```diff
@@ -1,54 +1,68 @@
...
+- `base_feature_extractor`: Base feature extractor for images. Defaults to ``pretrained_model_name_or_path`` if None.
+- `use_default_preprocessing`: Whether to use the `base_feature_extractor` on images (blobs) before encoding them, default is `True`.
+- `device`: Pytorch device to put the model on, e.g. 'cpu', 'cuda'.
+- `traversal_paths`: traversal path (use `cpu` if not specified in request's parameters).
+- `batch_size`: default batch size (use `32` if not specified in request's parameters).
+
+#### Inputs
```
Comment (cristianmtr) [inline jinahub/encoders/image/CLIPImageEncoder/README.md:17]: Same for inputs. Check https://github.com/jina-ai/executor-simpleindexer

**Review by cristianmtr — DISMISSED** [review]

File: jinahub/encoders/image/CLIPImageEncoder/README.md [hunk 3]
```diff
@@ -1,54 +1,51 @@
...
 
-It takes `Document`s with images stored in the `blob` attribute as inputs, and stores the
-resulting image embedding in the `embedding` attribute. You can store original images in
-the `blob` attribute, and they should be in the RGB format and have a shape `[H, W, 3]`). You
-can also choose to pass in already pre-processed images (see the class documentation).
-
-The **CLIP** model was originally proposed in [Learning Transferable Visual Models From Natural Language Supervision](https://arxiv.org/abs/2103.00020), and is trained to embed images and text to the same latent
-space. The corresponding text encoder is **[CLIPTextEncoder](https://hub.jina.ai/executor/livtkbkg)**,
-using both encoders together works well in multi-modal or cross-modal search applications.
+This encoder is meant to be used in conjunction with the [CLIPTextEncoder](https://github.com/jina-ai/executors/tree/main/jinahub/encoders/text/CLIPTextEncoder),
```
Comment (nan-wang) [inline jinahub/encoders/image/CLIPImageEncoder/README.md:5]: Use the link to the jina hub
```suggestion
This encoder is meant to be used in conjunction with the [CLIPTextEncoder](https://hub.jina.ai/executor/livtkbkg),

```

File: jinahub/encoders/image/CLIPImageEncoder/README.md [hunk 4]
```diff
@@ -1,54 +1,51 @@
...
+as it can embedd text and images to the same latent space.
```
Comment (nan-wang) [inline jinahub/encoders/image/CLIPImageEncoder/README.md:6]:
```suggestion
as it can embed text and images to the same latent space.

```

File: jinahub/encoders/image/CLIPImageEncoder/README.md [hunk 5]
```diff
@@ -1,54 +1,51 @@
...
+or in the .yml config.
 
-with f:
-    doc = Document(blob=np.asarray(Image.open('myimage.png').convert('RGB')))
-    f.post(on='/foo', inputs=doc, on_done=print_result)
+```yaml
+jtype: Flow
+pods:
+  - name: encoder
+    uses: 'jinahub+docker://AudioCLIPTextEncoder'
```
Comment (nan-wang) [inline jinahub/encoders/image/CLIPImageEncoder/README.md:26]:
```suggestion
    uses: 'jinahub+docker://CLIPImageEncoder'

```

**Review by nan-wang — CHANGES_REQUESTED** [review]

File: jinahub/encoders/image/CLIPImageEncoder/README.md (hunk 5 above)
Comment (bwanglzu) [reply jinahub/encoders/image/CLIPImageEncoder/README.md:27]: good catch!

File: jinahub/encoders/image/CLIPImageEncoder/README.md [hunk 6]
```diff
@@ -1,67 +1,17 @@
...
-Here's how you would modify the example above to use a GPU
-
-```python
-f = Flow().add(
-    uses='jinahub+docker://CLIPImageEncoder/gpu',
-    uses_with={'device': 'cuda'},
-    gpus='all',
-    volumes='/your/home/dir/.cache/huggingface:/root/.cache/huggingface' 
-)
+f = Flow().add(uses='jinahub+docker://CLIPImageEncoder',)
```
Comment (cristianmtr) [inline jinahub/encoders/image/CLIPImageEncoder/README.md:12]: same as in the other PR. Again, look at https://hub.jina.ai/executor/zb38xlt
Screenshot from 2021-09-17 16-10-06
4

**Review by nan-wang — APPROVED** [review]
LGTM👍
